src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Primitives
+{
+    public class CompositeChangeToken : IChangeToken
+    {
+        public CompositeChangeToken(IList<IChangeToken> changeTokens)
```

Comment:
`IReadOnlyList`? I think it should be fine requiring a list here and not an `IEnumerable`. Usually the `HasChanged` is queried frequently enough that we don't want to pay the cost of Enumerable allocations.